Log address

tescher opened the pull request:
Will log the street address geocoded from string input. Note that it only will log the last street in the case of an intersection entered by the user.

Resolves #62

NigelKibodeaux commented on lib/index.js:
```diff
@@ -80,6 +82,7 @@ function getStopFromBusTrackerId(busTrackerId, stopId, callback) {
             var parsed = {
                 stops: []
             };
+            console.log(body);
```
leftover debugging code?

tescher replied:
Yes, left it in there in case someone wanted to look at what was coming
through since there was some question about usefulness. It can be removed.
On Jan 16, 2016 9:12 PM, "Nigel Kibodeaux" <notifications@github.com> wrote:

> In lib/index.js
> <https://github.com/codeforanchorage/realtime-bus-sms/pull/66#discussion_r49940557>
> :
>
> > @@ -80,6 +82,7 @@ function getStopFromBusTrackerId(busTrackerId, stopId, callback) {
> >              var parsed = {
> >                  stops: []
> >              };
> > +            console.log(body);
>
> leftover debugging code?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/codeforanchorage/realtime-bus-sms/pull/66/files#r49940557>
> .[image: Web Bug from
> https://github.com/notifications/beacon/ACcF3EkQhNhWAPg6fm6ahiGuywTv5Qnfks5pav4igaJpZM4G9XoA.gif]
>

NigelKibodeaux replied:
cool, let's keep it if it's on purpose.

NigelKibodeaux commented on lib/index.js:
```diff
@@ -218,7 +221,8 @@ function getStopsFromAddress(address, callback) {
         lon = geocodedAddress.longitude;
-        console.log('geocoded "' + address + '" to: ' + lat + ', ' + lon);
+        // console.log('geocoded "' + address + '" to: ' + lat + ', ' + lon);
+        console.log('geocoded "' + address + '" to: ' + JSON.stringify(geocodedAddress));
```
leftover debugging code?

NigelKibodeaux commented on lib/index.js:
```diff
@@ -256,7 +261,7 @@ function processFeedback(feedback, callback, returnHtml) {
     var resp = 'Thanks for the feedback';
     resp += returnHtml ? ' <br> <a href="/">back</a>' : '';
-    callback(null, resp);
+    callback(null, resp, null, null, returnHtml);
```
I know it didn't happen in this branch, but this function doesn't need a callback since it's not asynchronous. We should just use a simpler synchronous function.

NigelKibodeaux commented on routes/index.js:
```diff
@@ -20,6 +20,29 @@ function logRequest(entry) {
 
+function sendIt(req, res, next, err, data, geocodedAddress, altInput, returnHtml) {
+    if (err) {
+        next(err)
```
should be `return next(err)` so the code doesn't continue

tescher commented:
Removed the console.log(body) but left in the geocoding log. Fixed "return next(err)".

As for the callbacks, I think I'd rather wait for a larger refactoring that needs to be done because it is getting a bit messy. Right now I'd like to add a testing suite (nodeunit) before I do that so that we can at least run smoke tests after big changes, if that's OK.

NigelKibodeaux commented:
yeah that sounds great

On Sun, Jan 17, 2016 at 8:58 AM tescher <notifications@github.com> wrote:

> Removed the console.log(body) but left in the geocoding log. Fixed "return
> next(err)".
>
> As for the callbacks, I think I'd rather wait for a larger refactoring
> that needs to be done because it is getting a bit messy. Right now I'd like
> to add a testing suite (nodeunit) before I do that so that we can at least
> run smoke tests after big changes, if that's OK.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/codeforanchorage/realtime-bus-sms/pull/66#issuecomment-172359133>
> .
>

tescher commented:
This can be merged now
